static/js/components/SingletonsContentListing.test.tsx

```diff
@@ -56,9 +56,9 @@ describe("SingletonsContentListing", () => {
     })
     website = makeWebsiteDetail()
     singletonConfigItems = [
-      makeSingletonConfigItem(),
-      makeSingletonConfigItem(),
-      makeSingletonConfigItem()
+      makeSingletonConfigItem("textId-1"),
+      makeSingletonConfigItem("textId-2"),
+      makeSingletonConfigItem("textId-3")
```

Comment:
Nice sleuthing figuring out that this was caused by duplicate names.

This change fixes the issue *for this test*, but it seems to me that makeSingletonConfigItem could create duplicate names else. Could we change `makeSingletonConfigItem` so that it always uses a unique name?

Maybe we could add a `uniqueWord` helper function, like
```ts
const uniqueWord = () => casual.word + _.uniqueId()
```
doesn't have to be [`_.uniqueId`](https://lodash.com/docs/3.10.1#uniqueId)...could be anything to guarantee uniqueness.

_Random aside: I've recently been prefering `@faker-js/faker` for generating test data, largely for this reason. It has a [`faker.helpers.unique`](https://fakerjs.dev/api/helpers.html#unique) method. Not sure that's worth adding to studio just for this, though._

Reply:
@ChristopherChudzicki Thank you. I agree. I did think about this option.

I chose to keep it simple for two reasons:
1. This change impacts the least amount of code. Makes it easier to test, and track bugs caused by it.
2. If other tests already pass with duplicates, we should let them be. In this case, removing duplicates only reduces our test data space.

I don't know what a "SingletonConfigItem" really is. If it is meant to always have a unique name property then my opinion changes and I think your suggestion is better.

Let me know what you think.

PS: Yeah, faker is great, it's my first choice too.

Reply:
Re "what a "SingletonConfigItem" really is": OCW Studio's behaves differently for different types of sites (example: courses and ocw-www). The behavior is controlled via config files which are stored in [ocw-hugo-projects](https://github.com/mitodl/ocw-hugo-projects).

RepeatableConfigItem is the configuration for "repeatable" content, like pages or resources. (example: pages https://github.com/mitodl/ocw-hugo-projects/blob/main/ocw-course-v2/ocw-studio.yaml#L6).

SingletonConfigItem is configuration for singleton content, like the "Menu" (https://github.com/mitodl/ocw-hugo-projects/blob/main/ocw-course-v2/ocw-studio.yaml#L777). *"Menu" is the links that are on the left-hand side of OCW courses*.

--- 

Re
> I chose to keep it simple for two reasons...

Makes sense. IMO, reason (2) is not a concern: `name` should be unique. If it's not unique, we will have lots of problems.

Re (1), since `makeSingletonConfigItem` is only used in tests, if the tests pass, I think you're safe. Especially if they pass 100 times with your loop :).